pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js

```diff
@@ -0,0 +1,28 @@
+// Code provided from Google reCAPTCHA
+// https://www.sfdcpoint.com/salesforce/salesforce-web-to-lead-form-with-recaptcha/
+function timestamp() {
+  var response = document.getElementById("g-recaptcha-response");
+  if (response === null || response.value.trim() === "") {
+    var elems = JSON.parse(
+      document.getElementsByName("captcha_settings")[0].value
+    );
+    elems["ts"] = JSON.stringify(new Date().getTime());
+    document.getElementsByName("captcha_settings")[0].value = JSON.stringify(
+      elems
+    );
+  }
+}
+setInterval(timestamp, 500);
+
+// Set a disabled attribute on button until
+// the reCAPTCHA checkbox is clicked.
+// Uses the data-callback attribute to check for a successful response.
+// Docs: https://developers.google.com/recaptcha/docs/display
+// eslint-disable-next-line no-unused-vars
+function recaptchaCallback() {
+  const btnSubmit = document.querySelector("button");
+
+  if (btnSubmit.hasAttribute("disabled")) {
+    btnSubmit.removeAttribute("disabled");
+  }
+}
```

Comment:
one thing to note is that the linter rules for this file are coming from this laundry list of possible rules here: https://github.com/code-dot-org/code-dot-org/blob/0579d01daf5b014c54b1dcf41d3a2e328cfd9f34/.eslintrc.js if we start adding a lot more js in pegasus, we should consider getting more deliberate about what rules we want to enable in the public/js directories in pegasus, possibly including a predefined set like es5.

on the other hand, if most new js is going to be in react / in the apps/src directory then we are good. we get more specific in the apps directory where we specifically allow es6: https://github.com/code-dot-org/code-dot-org/blob/2d4125fc9b98f0d308240544579bb8f717e25be9/apps/.eslintrc.js#L14-L15 which we can do because we are running all that js through [babel](https://babeljs.io/).

food for thought 🥗 🧠

Reply:
I think it would be good to add more js in general to pegasus to make some stuff have better UX and/or delight as we keep redesigning pages 😄